### ISSUE TITLE
java.lang.SecurityException caused by jar signer information

### ISSUE THREAD

**Issue by TTJudy** [lightbody/browsermob-proxy#470]
Here is my code:
`BrowserMobProxy proxy = new BrowserMobProxyServer();`
`proxy.start(0);`
`Proxy seleniumProxy = ClientUtil.createSeleniumProxy(proxy);`
The code builds on Windows machines, but fails on Linux with the following exception (the exception happens on the BrowserMobProxyServer initiation):

> Caused by: java.lang.SecurityException: class "org.bouncycastle.jcajce.provider.symmetric.IDEA$Mappings"'s signer information does not match signer information of other classes in the same package
	at java.lang.ClassLoader.checkCerts(ClassLoader.java:952)
	at java.lang.ClassLoader.preDefineClass(ClassLoader.java:666)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:794)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
	at org.bouncycastle.jce.provider.BouncyCastleProvider.loadAlgorithms(Unknown Source)
	at org.bouncycastle.jce.provider.BouncyCastleProvider.setup(Unknown Source)
	at org.bouncycastle.jce.provider.BouncyCastleProvider.access$000(Unknown Source)
	at org.bouncycastle.jce.provider.BouncyCastleProvider$1.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at org.bouncycastle.jce.provider.BouncyCastleProvider.<init>(Unknown Source)
	at net.lightbody.bmp.mitm.tools.BouncyCastleSecurityProviderTool.<clinit>(BouncyCastleSecurityProviderTool.java:63)
	... 12 more

Can you please shed some light on this? Thank you!

**Comment by TTJudy** [comment]
To provide you with more information:
My Windows & Linux box have the same jdk version.

**Comment by yotam-supersonic** [comment]
Are you using selenium-server-standalone? I had this problem until I got rid of this one.

**Comment by TTJudy** [comment]
@yotam-supersonic Yes I'm using selenium-server-standalone-2.47.0. What other selenium library did you use then?

**Comment by yotam-supersonic** [comment]
Well turns out I didn't need it, so nothing. It was and old leftover.

**Comment by jekh** [comment]
It looks to me like a conflict between some other version of bouncycastle on your classpath and the recent version BMP requires. You may need to dig into your dependency tree to figure out what is pulling in the older bouncycastle version, and upgrade that to a more recent version.

**Comment by dmhalejr** [comment]
@jekh 

I've looked into this issue trying to get `BMP 2.1.2` running with `Selenium 3.0.0-beta3` and took your advice to look into my dependency tree and found two elements coming directly from BMP. I don't have any additional dependency on `bouncycastle` other then what is currently available in `2.1.2`. 

Should this commit mediate the issue or does the error still live on my end? 
https://github.com/lightbody/browsermob-proxy/commit/d15f7612ea872819f2ab5b23aa0d9ec3febdbba5

System: 
```
OS X 10.11.5 (15F34)

java version "1.8.0_74"
Java(TM) SE Runtime Environment (build 1.8.0_74-b02)
Java HotSpot(TM) 64-Bit Server VM (build 25.74-b02, mixed mode)
```

Dependency Tree:
```
➞  ./gradlew dependencyInsight --dependency org.bouncycastle                                                                                                                          
:dependencyInsight
org.bouncycastle:bcpkix-jdk15on:1.54
+--- net.lightbody.bmp:browsermob-core:2.1.2
|    \--- compile
\--- net.lightbody.bmp:mitm:2.1.2
     \--- net.lightbody.bmp:browsermob-core:2.1.2 (*)

org.bouncycastle:bcprov-jdk15on:1.54
+--- net.lightbody.bmp:browsermob-core:2.1.2
|    \--- compile
+--- net.lightbody.bmp:mitm:2.1.2
|    \--- net.lightbody.bmp:browsermob-core:2.1.2 (*)
\--- org.bouncycastle:bcpkix-jdk15on:1.54
     +--- net.lightbody.bmp:browsermob-core:2.1.2 (*)
     \--- net.lightbody.bmp:mitm:2.1.2 (*)
```

**Comment by jekh** [comment]
@dmhalejr - What's the error you're seeing? I'd be interested to know if updating the bc version does solve the issue for your, since the 1.55 update had little (if anything) that should actually affect BMP.

Good idea to test BMP with the Selenium 3 beta, though!
